### PR TITLE
Merge master into features/ioperation

### PR DESCRIPTION
**Pull request by mavasani**

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @analyze-ioperation @heejaechang for review

**Comment by mavasani** [comment]
@analyzer-ioperation

**Comment by mavasani** [comment]
Tagging @dotnet/analyzer-ioperation

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 1]
```diff
@@ -167,10 +168,29 @@ private static IOperation CreateInternal(BoundNode boundNode)
...
 
+        private static ImmutableArray<IOperation> GetIOperationChildren(BoundNode boundNode)
+        {
+            var boundNodeWithChildren = (IBoundNodeWithIOperationChildren)boundNode;
+            if (boundNodeWithChildren.Children.IsDefaultOrEmpty)
+            {
+                return ImmutableArray<IOperation>.Empty;
+            }
+
+            var builder = ArrayBuilder<IOperation>.GetInstance(boundNodeWithChildren.Children.Length);
```
Comment (heejaechang) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: boundNodeWithChilred.Children.SelectAsArray(n => Create(n)); 

might be about same perf?

**Comment by heejaechang** [comment]
does this contain your change + regular FI chagnes from master?

File: src/Compilers/VisualBasic/Portable/Binding/MemberSemanticModel.vb [hunk 2]
```diff
@@ -800,7 +800,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     result = summary.LowestBoundNode
             End Select
 
-            Return TryCast(result, IOperation)
+            Return VisualBasicOperationFactory.Create(result)
```
Comment (heejaechang) [inline src/Compilers/VisualBasic/Portable/Binding/MemberSemanticModel.vb]: sorry about this.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 3]
```diff
@@ -164,10 +164,26 @@ Namespace Microsoft.CodeAnalysis.Semantics
...
             End Select
         End Function
 
+        Private Shared Function GetIOperationChildren(boundNode As BoundNode) As ImmutableArray(Of IOperation)
+            Dim boundNodeWithChildren = DirectCast(boundNode, IBoundNodeWithIOperationChildren)
+            If boundNodeWithChildren.Children.IsDefaultOrEmpty Then
+                Return ImmutableArray(Of IOperation).Empty
+            End If
+
+            Dim builder = ArrayBuilder(Of IOperation).GetInstance(boundNodeWithChildren.Children.Length)
```
Comment (heejaechang) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: same as above SelectAsArray might do same perf.

**Comment by heejaechang** [comment]
overall looks fine. thank you for finding VB issue. not sure what happened.

it would have been nice if FI changes and your change could have separated.

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs (hunk 1 above)
Comment (mavasani) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Compiler team prefers this pattern of code over linq.

**Comment by mavasani** [comment]
@heejaechang This includes FI as well as minor fixes to VB factory, and there is a separate commit to track it. I have skipped bunch of tests with https://github.com/dotnet/roslyn/pull/19348 for you to fix - please search for tests skipped with 19348 in VB semantic tests project.

**Comment by mavasani** [comment]
@dotnet-bot retest windows_release_vs-integration_prtest please

**Pull request merged**